### PR TITLE
Implement Jest testing platform for all exercises

### PR DESCRIPTION
**Pull request by xandora**
This PR overhauls the JavaScript exercises to use Jest instead of Jasmine.

- Updated README.md content.
- Updated syntax to be consistent across exercises.
- Users can simply `npm install` to pull required node modules in using `package.json` file.
- ESLint has been included, but is set to run manually as a forward thinking possibility for a "Linting lesson" using existing lessons.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,3 @@
 .vscode
-node_modules
-package-lock.json
-package.json
\ No newline at end of file
+node_modules/
+.DS_Store
```
Comment (twalton83) [inline .gitignore:3]: Please add a new line, thank you!

File: calculator/calculator.js [hunk 2]
```diff
@@ -1,32 +1,32 @@
...
-	factorial
-}
\ No newline at end of file
+  add,
+  subtract,
+  sum,
+  multiply,
+  power,
+  factorial
+};
```
Comment (twalton83) [inline calculator/calculator.js:32]: Need a new line here too, thank you!

File: fibonacci/README.md [hunk 3]
```diff
@@ -1,8 +1,8 @@
-# Exercise XX - fibonacci
+# Exercise XX - Fibonacci
```
Comment (twalton83) [inline fibonacci/README.md:1]: I appreciate you capitalizing the exercises. Can we get the actual exercise number here? Thanks!

File: pigLatin/pigLatin.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+function pigLatin(string) {
+
+};
+  
+  module.exports = pigLatin;
```
Comment (twalton83) [inline pigLatin/pigLatin.js:5]: New line me!

**Review by twalton83 — CHANGES_REQUESTED** [review]
Just a few new line comments and small things, thank you for your hard work on this!

File: pigLatin/pigLatin.js (hunk 4 above)
Comment (xandora) [reply pigLatin/pigLatin.js]: Done.

File: fibonacci/README.md (hunk 3 above)
Comment (xandora) [reply fibonacci/README.md]: Done.

File: calculator/calculator.js (hunk 2 above)
Comment (xandora) [reply calculator/calculator.js]: Done.

File: .gitignore (hunk 1 above)
Comment (xandora) [reply .gitignore:3]: Done.

**Review by twalton83 — APPROVED** [review]
LGTM

**Comment by twalton83** [comment]
Closing because this will be addressed with #149 instead

**Pull request closed without merging**